contracts/BaseStrategy.sol

```diff
@@ -720,14 +726,13 @@ abstract contract BaseStrategy {
         uint256 debtPayment = 0;
         if (emergencyExit) {
             // Free up as much capital as possible
-            uint256 totalAssets = estimatedTotalAssets();
-            // NOTE: use the larger of total assets or debt outstanding to book losses properly
-            (debtPayment, loss) = liquidatePosition(totalAssets > debtOutstanding ? totalAssets : debtOutstanding);
-            // NOTE: take up any remainder here as profit
-            if (debtPayment > debtOutstanding) {
-                profit = debtPayment.sub(debtOutstanding);
-                debtPayment = debtOutstanding;
+            uint256 amountFreed = liquidateAllPositions();
```

Comment:
@fubuloubu wondering if we should use the returned value or use `want.balanceOf(address(this));`

Reply:
shouldn't `amountFreed := debtPayment + profit - loss`?

Reply:
We should not add gain and debtPayment
This line will fail.
https://github.com/steffenix/yearn-vaults/blob/c818595d1699077d20285c5fdd28ed2a9f564b6c/contracts/Vault.vy#L1681